integration-tests/virtual-threads/graphql-virtual-threads/src/test/java/io/quarkus/virtual/graphql/RunOnVirtualThreadTest.java

```diff
@@ -0,0 +1,170 @@
+package io.quarkus.virtual.graphql;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit5.virtual.ShouldNotPin;
+import io.quarkus.test.junit5.virtual.VirtualThreadUnit;
+import io.quarkus.virtual.threads.VirtualThreads;
+import io.restassured.RestAssured;
+import io.smallrye.common.annotation.RunOnVirtualThread;
+import io.vertx.core.Vertx;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.graphql.GraphQLApi;
+import org.eclipse.microprofile.graphql.Mutation;
+import org.eclipse.microprofile.graphql.Query;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.ExecutorService;
+
+@QuarkusTest
+@VirtualThreadUnit
+@ShouldNotPin
+class RunOnVirtualThreadTest extends AbstractGraphQLTest {
+
+    //todo how to make sure vt executor is not removed?
```

Comment:
Maybe @cescoffier knows - do we have any general mechanism to detect usage of virtual threads?

At minimum though, in case we are to keep this annotation, I would move it into the `RunOnVirtualThreadObjectTestThreadResource` subclass because then it won't prevent running this test in native mode

Reply:
I moved it to the subclass.